presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1116,6 +1116,8 @@ core::JoinType toJoinType(protocol::JoinType type) {
       return core::JoinType::kRight;
     case protocol::JoinType::FULL:
       return core::JoinType::kFull;
+    case protocol::JoinType::SOURCE_OUTER:
+      return core::JoinType::kInner; // TODO: Map to proper join type.
```

Comment:
Can we fail this for now?

Reply:
I think it's needed. There's test requires all join type to be available in the protocol: https://github.com/prestodb/presto/pull/24768#issuecomment-2786687972